core-referrals/lib/core/referrals/railtie.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+class String
```

Comment:
Since we are touching some of the code across projects in this PR, is this an opportunity to move the utility extension into core-base, along with a helper for calling the DX checks?

Reply:
I thought about doing it but felt pressed for time. I'll look at it 👍

Reply:
Added more cleanup and moved the `String` additions to core-base only, but I'm not completely convinced the initializer sections aren't app specific 🤔